rust/index/src/spann/types.rs

```diff
@@ -742,6 +847,42 @@ impl SpannIndexWriter {
                 next_version,
                 prev_head_id
             );
+            self.metrics.num_reassigns.add(
+                1,
+                &[KeyValue::new(
+                    "collection_id",
+                    self.collection_id.to_string(),
```

Comment:
to_string on this will add up. Suggest you cache this intermediate value given the frequency.